iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/InjectionObservabilityAdder.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.extensions;
+
+import com.powsybl.commons.extensions.ExtensionAdder;
+import com.powsybl.iidm.network.Injection;
+
+/**
+ * @author Thomas Adam <tadam at silicom.fr>
+ */
+public interface InjectionObservabilityAdder <I extends Injection<I>>
+        extends ExtensionAdder<I, InjectionObservability<I>> {
+
+    @Override
+    default Class<InjectionObservability> getExtensionClass() {
+        return InjectionObservability.class;
+    }
+
+    InjectionObservabilityAdder<I> withObservable(boolean observable);
+
+    InjectionObservabilityAdder<I> withStandardDeviationP(double standardDeviationP);
+
+    InjectionObservabilityAdder<I> withRedundantP(boolean redundant);
+
+    InjectionObservabilityAdder<I> withStandardDeviationQ(double standardDeviationQ);
+
+    InjectionObservabilityAdder<I> withRedundantQ(boolean redundant);
+
+    InjectionObservabilityAdder<I> withStandardDeviationV(double standardDeviationV);
+
+    InjectionObservabilityAdder<I> withRedundantV(boolean redundant);
```

Comment:
I find it a bit disturbing to have different structures in the adder and the built object (but maybe this is just me, do you have an opinion @sylvlecl @flo-dup ?). Both options (an `ObservabilityQuality` sub-object or only parameters) are good to me (I think I prefer the second option though), I would just would rather for the APIs to be homogeneous.

Reply:
No strong opinion either, but it looks OK to me the way it is:
- the builder API seems simpler to me that way, both for the user and for the maintainer (not having sub-builders or something like that which would make it complicated)
- on the other hand, having sub-objects in the built object allows for some code sharing and just makes sense from the business/model point of view